src/meta/src/manager/cluster.rs

```diff
@@ -225,47 +226,28 @@ where
 
     pub async fn update_schedulability(
         &self,
-        worker_ids: &[u32],
-        is_unschedulable: bool,
+        worker_ids: Vec<u32>,
+        schedulability: Schedulability,
     ) -> MetaResult<()> {
+        let worker_ids: HashSet<_> = worker_ids.into_iter().collect();
+
         let mut core = self.core.write().await;
-        let mut workers = core
-            .workers
-            .values_mut()
-            .filter(|w| worker_ids.contains(&w.worker_id()))
-            .collect_vec();
-        if workers.len() != worker_ids.len() {
-            return Err(MetaError::invalid_parameter(format!(
-                "Tried to update schedulability of workers {:?}. Only found workers {:?}",
-                worker_ids,
-                workers.iter().map(|w| w.worker_id()).collect_vec()
-            )));
-        }
-        let workers_need_change = workers
-            .iter_mut()
-            .filter(|w| match w.worker_node.property.as_ref() {
-                None => true,
-                Some(p) => p.is_unschedulable != is_unschedulable,
-            })
-            .collect_vec();
-        if workers_need_change.is_empty() {
-            return Ok(());
-        }
 
-        if workers_need_change
-            .iter()
-            .any(|w| w.worker_node.property.is_none())
-        {
-            return Err(MetaError::invalid_parameter(
-                "Worker node does not have property",
-            ));
+        let mut txn = Transaction::default();
+        for worker in core.workers.values_mut() {
+            if worker_ids.contains(&worker.worker_node.id) {
+                if let Some(property) = worker.worker_node.property.as_mut() {
+                    let target = schedulability == Schedulability::Unschedulable;
+                    if property.is_unschedulable != target {
+                        property.is_unschedulable = target;
```

Comment:
Should get it updated after txn success, otherwise it will cause inconsistency.

Reply:
Thanks! I have rewritten it using VarTransaction.